### PR TITLE
ttl: Add some tools to build SQLs for TTL

### LINKED ISSUE #39339: Add tools to generate TTL SELECT/DELETE SQLs

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #39339

### What is changed and how it works?

Add some tools to generate SQL of the TTL:

- Add `ttl.SQLBuilder` to provide some common methods to build a SELECT/DELETE SQL
- Add `ttl.ScanQueryGenerator` to generate scan task SQLs.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

File: ttl/sql.go [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+	writeWhere
+	writeOrderBy
+	writeLimit
+	writeDone
+)
+
+// SQLBuilder is used to build SQLs for TTL
+type SQLBuilder struct {
+	tbl   *PhysicalTable
+	sb    strings.Builder
```
Comment (xhebox) [inline ttl/sql.go:76]: Maybe you should consider https://github.com/pingcap/tidb/blob/d100e933c8982ed13135abbca1224b472b167b53/util/sqlexec/utils.go#L247
if there is a risk of injection.

Comment (lcwangchao) [reply ttl/sql.go:76]: I'm using `sqlexec.EscapeString` to escape the value. Seems they are equivalent in implement

Comment (xhebox) [reply ttl/sql.go:76]: Sure , but table name, or every identifier may have a problem. BTW, `FormatSQL` is the builder version of `EscapeSQL`. Check again if you really need escaping everywhere or just datum.

**Review by xhebox — APPROVED** [review]

**Comment by lcwangchao** [comment]
> Sure , but table name, or every identifier may have a problem. BTW, FormatSQL is the builder version of EscapeSQL. Check again if you really need escaping everywhere or just datum.

@xhebox Yes, you are right. We still to escape table, partition and column names. I've updated the PR and added some tests.

File: ttl/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        "//parser/model",
+        "//parser/mysql",
+        "//testkit",
+        "//testkit/testsetup",
+        "//types",
+        "//util/sqlexec",
+        "@com_github_stretchr_testify//require",
+        "@org_uber_go_goleak//:goleak",
+    ],
+)
```
Comment (hawkingrei) [inline ttl/BUILD.bazel:41]:
```suggestion
go_test(
    name = "ttl_test",
    srcs = [
        "main_test.go",
        "sql_test.go",
    ],
    deps = [
        ":ttl",
        "//kv",
        "//parser/model",
        "//parser/mysql",
        "//testkit",
        "//testkit/testsetup",
        "//types",
        "//util/sqlexec",
        "@com_github_stretchr_testify//require",
        "@org_uber_go_goleak//:goleak",
    ],
    flaky = True,
)
```

Comment (hawkingrei) [reply ttl/BUILD.bazel:41]: add ```flaky = True,```

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**